pykechain/models/customization.py

```diff
@@ -167,6 +167,111 @@ def add_json_widget(self, config):
+        :type max_height: int or None
+        :param custom_title: A custom title for the supergrid
+                 - None (default): Part instance name
+                 - String value: Custom title
+                 - False: No title
+        :type custom_title: basestring or None
+        """
+        # Assertions
+        if not (new_instance and bool(parent_part_instance)):
+            raise IllegalArgumentError("If you want to allow the creation of new part instances, you must specify a "
+                                       "parent_part_instance")
+
+        # Declare supergrid config
+        config = {
+            "xtype": ComponentXType.SUPERGRID,
+            "filter": {
+                "activity_id": str(self.activity.id),
+                "model": str(part_model.id)
+            },
+            "viewModel": {
+                "data": {
+                    "actions": {
+                        "delete": delete,
+                        "edit": edit,
+                        "export": export,
+                        "incompleteRows": incomplete_rows,
+                        "newInstance": new_instance
+                    },
+                    "ui": {
+                        "newInstance": "primary-action" if emphasize_new_instance else "default-toolbar",
+                        "edit": "primary-action" if emphasize_edit else "default-toolbar"
+                    }
+                }
+            }
+        }
+
+        # Add parent to filter if added.
+        if parent_part_instance:
+            config['filter']["parent"] = str(parent_part_instance.id)
+
+        # Add max height and custom title
+        if max_height:
+            config['maxHeight'] = max_height
+
+        if custom_title is None:
+            show_title_value = "Default"
+            title = part_model.name
+        elif custom_title:
+            show_title_value = "Custom Title"
+            title = str(custom_title)
+        else:
+            show_title_value = "No title"
+            title = None
+        config["title"] = title
+
+        # Declare the meta info for the supergrid
+        meta = {
+            "parentInstanceId": str(parent_part_instance.id),
+            "editButtonUi": "primary-action" if emphasize_edit else "default-toolbar",
+            "customHeight": max_height if max_height else 500,
+            "primaryAddUiValue": emphasize_new_instance,
+            "activityId": str(self.activity.id),
+            "customTitle": title,
+            "primaryEditUiValue": emphasize_edit,
+            "downloadButtonVisible": export,
+            "addButtonUi": "primary-action" if emphasize_new_instance else "default-toolbar",
+            "deleteButtonVisible": delete,
+            "incompleteRowsButtonVisible": incomplete_rows,
+            "addButtonVisible": new_instance,
+            "showTitleValue": show_title_value,
+            "partModelId": str(part_model.id),
```

Comment:
I would make sure you can inject an uuid as well.. see comments above with example code